### ISSUE TITLE
Fix 32-bit Windows compile error

### ISSUE THREAD

**Issue by andreastt** [mozilla/geckodriver#246]
https://travis-ci.org/mozilla/geckodriver/builds/163755698

**Comment by andreastt** [comment]
Filed https://github.com/alexcrichton/port-of-rust/issues/5.

**Comment by andreastt** [comment]
The port-of-rust Docker image we are building is using Nightly which has [a bug](https://github.com/rust-lang/rust/issues/36793#issuecomment-2508036770) causing the i686 Windows build to fail.

Unfortunately Rust stable does not have the i686-pc-windows-gnu target, and the the same bug that is in Nightly affects Beta.

I think the short-term solution here is to fork the [Dockerfile](https://github.com/alexcrichton/port-of-rust/blob/master/i686-pc-windows-gnu/Dockerfile) and pin it to a known-good version of Rust (probably beta by this point), so that we can make a release early next week.

**Comment by andreastt** [comment]
It looks like the bug in nightly was fixed in https://github.com/rust-lang/rust/pull/36866, but there are build issues with Rust for iOS, which means there have been no nightly builds yet with this fix in it.

**Comment by andreastt** [comment]
It looks like cross-compilation for 64-bit Windows is also failing, but isn’t surfacing on geckodriver’s master builds because they use a cached Rust Nightly version which is known to be good:

```
+rustc -V
rustc 1.13.0-beta.1 (cbbeba430 2016-09-28)
```

https://travis-ci.org/andreastt/geckodriver/builds/164048072

**Comment by andreastt** [comment]
The x86_64-pc-windows-gnu cross-compile also fails on stable: https://travis-ci.org/andreastt/geckodriver/jobs/164685055

**Comment by andreastt** [comment]
Right, after many hours of digging I’ve found that this is probably a regression Travis, our CI. We rely on the container infrastructure they provide to get faster builds. Up until around 2016-09-21 the containers provided were running Ubuntu trusty, the distribution version we had requested. Sometime in the days after this Travis started providing Ubuntu precise container images on a trusty host OS.

Ubuntu precise obviously have a much older gcc version than trusty, and linking probably fails because it produces something Rust can’t use. I confirmed this through using a custom container on Travis, which fixes our 64-bit build.

The 32-bit build is failing because of https://github.com/rust-lang/rust/pull/36866, but we should be able to circumvent this by pinning the Rust Nightly version like this:

    rustup override set nightly-2016-09-21

Which also, I found, can be used for the rustup installer:

    curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain=nightly-2016-09-21
